### ISSUE TITLE
Websockets 4.0 does not work with sanic master

### ISSUE THREAD

**Issue by dsanghan** [channelcat/sanic#1009]
Socket connections break with `TypeError: A Future, a coroutine or an awaitable is required`. Might be a uvloop bug but anyone installing sanic via pip3 won't be able to use websockets. 

Websockets 3.4 works fine.

**Comment by niobos** [comment]
Also, the included example (`examples/websocket.py`) doesn't work anymore (works with `websockets==3.4`, fails with `websockets==4.0.1`), probably because of a related bug:

```
2017-11-16 08:55:50 - (sanic)[INFO]: Goin' Fast @ http://0.0.0.0:8000
2017-11-16 08:55:50 - (sanic)[INFO]: Starting worker [2588]
2017-11-16 08:55:55 - (network)[INFO][127.0.0.1:54997]: GET http://localhost:8000/  200 1141
Sending: hello!
2017-11-16 08:55:55 - (sanic)[ERROR]: Traceback (most recent call last):
  File "/private/tmp/sanic/venv/lib/python3.6/site-packages/sanic/app.py", line 503, in handle_request
    response = await response
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/coroutines.py", line 128, in throw
    return self.gen.throw(type, value, traceback)
  File "/private/tmp/sanic/venv/lib/python3.6/site-packages/sanic/app.py", line 247, in websocket_handler
    await fut
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/coroutines.py", line 125, in send
    return self.gen.send(value)
  File "websocket.py", line 17, in feed
    await ws.send(data)
  File "/private/tmp/sanic/venv/lib/python3.6/site-packages/websockets/protocol.py", line 334, in send
    yield from self.ensure_open()
  File "/private/tmp/sanic/venv/lib/python3.6/site-packages/websockets/protocol.py", line 484, in ensure_open
    raise InvalidState("WebSocket connection isn't established yet")
websockets.exceptions.InvalidState: WebSocket connection isn't established yet

2017-11-16 08:55:55 - (sanic)[ERROR]: Exception occurred while handling uri: "ws://localhost:8000/feed"
Traceback (most recent call last):
  File "/private/tmp/sanic/venv/lib/python3.6/site-packages/sanic/app.py", line 503, in handle_request
    response = await response
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/coroutines.py", line 128, in throw
    return self.gen.throw(type, value, traceback)
  File "/private/tmp/sanic/venv/lib/python3.6/site-packages/sanic/app.py", line 247, in websocket_handler
    await fut
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/coroutines.py", line 125, in send
    return self.gen.send(value)
  File "websocket.py", line 17, in feed
    await ws.send(data)
  File "/private/tmp/sanic/venv/lib/python3.6/site-packages/websockets/protocol.py", line 334, in send
    yield from self.ensure_open()
  File "/private/tmp/sanic/venv/lib/python3.6/site-packages/websockets/protocol.py", line 484, in ensure_open
    raise InvalidState("WebSocket connection isn't established yet")
websockets.exceptions.InvalidState: WebSocket connection isn't established yet
```

**Comment by danielgtaylor-isp** [comment]
This appears to be fixed from #1035 and is in the `0.7.0` release.

**Comment by r0fls** [comment]
@dsanghan please feel free to reopen this if you're still having any problems.
